src/node/tests/adult.rs

```diff
@@ -55,6 +55,7 @@ impl Env {
         let (subject, ..) = Node::approved(
             NodeConfig {
                 network_params: NETWORK_PARAMS,
+                rng: rng::new_from(&mut rng),
```

Comment:
Just curious: why was this needed?

Reply:
previously the adult's rng was using default rng, which is generated from local thread, which means it has the same seed as to the main rng within the testing code generating elder ids. 
This results in the adult's id is duplicated to one of the generated elder id, which fails the `is_elder` check, i.e. the adult think itself as an elder.